packages/search-service/src/helpers.ts

```diff
@@ -59,27 +85,29 @@ class IndexHelper {
         } );
     }
 
-    public search ( params: any ) {
+    public async search ( params: any ) {
+        const authData: any = await this.auth();
         return new Promise( async ( resolve, reject ) => {
             let headers = new Headers();
+            headers.append( `Authorization`, `Bearer ${ authData.access_token }` );
             headers.append( `Content-Type`, `application/json` );
             headers.append( `Accept`, `application/json` );
             try {
```

Comment:
Use .catch instead of try-catch here.
try catch is useful only when the code inside it is synchronous, or if it uses async await. Otherwise, it leads to unhandled exceptions.

Reply:
Also, there's no need to use `new Promise` here since your return statement already is a Promise.

Reply:
Updated everything as requested.